src/itype.h

```diff
@@ -902,6 +908,9 @@ struct itype {
         /** Magazine types (if any) for each ammo type that can be used to reload this item */
         std::map< ammotype, std::set<itype_id> > magazines;
 
+        /** Battery types (if any) that can be used to reload this item */
+        std::set<itype_id> batteries = {};
```

Comment:
```suggestion
        std::set<itype_id> batteries;
```

Reply:
Initializing it to an empty set is deliberate so I can check `something.batteries.empty()` to determine if an item uses batteries.

Reply:
```cpp
std::set<itype_id> batteries;
```
Calls the default constructor which creates an empty set.